recipes/libmysqlclient/all/conanfile.py

```diff
@@ -117,6 +117,8 @@ def validate(self):
             raise ConanInvalidConfiguration("GCC 5.3 or newer is required")
         if self.settings.compiler == "clang" and Version(self.settings.compiler.version) < "6":
             raise ConanInvalidConfiguration("clang 6 or newer is required")
+        if tools.cross_building(self, skip_x64_x86=True):
```

Comment:
```suggestion
        if hasattr(self, "settings_build") and tools.cross_building(self, skip_x64_x86=True):
```

Remember that we are computing all the packageIDs in a Linux node, and the helper `tools.cross_building` will infer the _build_ context from the running machine if no `settings_build` is available... so all the Windows and MacOS configurations are discarded (check the tables with generated binaries).

Reply:
Can't the bot set the proper profile in order to identify the package id's?

Reply:
The reason this is failing is because of this method in conan:
https://github.com/conan-io/conan/blob/916cb10b99ed92d319cd0719462708ee0501ecd4/conans/client/tools/oss.py#L500-L510
, more specifically these 2 lines:
```python
        os_build = self_os or os_build or detected_os()
        arch_build = self_arch or arch_build or detected_architecture()
```

- `self_os` is `None` because we don't pass an os to `tools.cross_building`
- `os_build` is `None` because `ConanFile.settings.os_build` is None/deprecated
- `detected_os()` always returns `Linux` on the linux master node

Ideally, this should be fixed in conan/c3i instead of our recipes.
**How to fix it in conan?**
First try getting `os_build`/`arch_build` from the profile, else detect it

**How to fix it in c3i?**
Always use 2 profile cross build approach. With host profile == build profile.

Reply:
> How to fix it in c3i?
> Always use 2 profile cross build approach. With host profile == build profile.

Everything is connected :wink:  --> This was one of the reasons why we introduced Apple M1 as a cross-compiling scenario, because it will force us to start using two profiles for that configuration and fix some broken recipes... and also prepare the CI library.

Now we would love to hear from people running M1 and know that the package ID is always found and available, that the same packageID is computed using one or two profiles ([remember here](https://github.com/conan-io/conan-center-index/pull/6686#issuecomment-893354790)) (or maybe a bug in Conan).

So far, people are not opening issues about M1 (sounds good), but we don't know if they are using these binaries (hopefully, @danimtb will be able to tell us something about it soon).

Once we are sure, we will start to use two profiles for all the configurations and we can get rid of these very inconvenient patches. Once we do that, it can also happen that recipes will stop working using one single profile, so we need to decide if we want to keep running some configurations with one profile, or we just propose https://github.com/conan-io/conan/issues/9336 as the recommended default (not Conan v1.x default) if you want to build CCI recipes from sources.